Assets/Editor/ModelInspector.cs

```diff
@@ -414,15 +488,102 @@ static Color GetSubMeshTint(int index)
                 else
                     previewUtility.DrawMesh(mesh, pos, rot, settings.wireMaterial, meshSubset, customProperties);
                 previewUtility.Render();
+                
                 GL.wireframe = false;
             }
 
             Unsupported.SetRenderSettingsUseFogNoDirty(oldFog);
         }
+        
+        public static Vector3 GetPosition(Matrix4x4 matrix)
+        {
+            Vector3 position;
+            position.x = matrix.m03;
+            position.y = matrix.m13;
+            position.z = matrix.m23;
+            return position;
+        }
+ 
+        public static Vector3 GetScale(Matrix4x4 matrix)
+        {
+            Vector3 scale;
+            scale.x = matrix.GetColumn(0).magnitude;
+            scale.y = matrix.GetColumn(1).magnitude;
+            scale.z = matrix.GetColumn(2).magnitude;
+            return scale;
+        }
+        
+        static void SetTransformMatrix(Transform tr, Matrix4x4 mat)
+        {
+            // extract position
+            var pos = new Vector3(mat.m03, mat.m13, mat.m23);            
+            
+            // extract scale
+            var scale = mat.lossyScale;            
+            
+            // now remove scale from the matrix axes,
+            var invScale = new Vector3(1.0f/scale.x, 1.0f/scale.y, 1.0f/scale.z);
+            mat.m00 *= invScale.x; mat.m10 *= invScale.x; mat.m20 *= invScale.x;
+            mat.m01 *= invScale.y; mat.m11 *= invScale.y; mat.m21 *= invScale.y;
+            mat.m02 *= invScale.z; mat.m12 *= invScale.z; mat.m22 *= invScale.z;
+            
+            // and extract rotation
+            var rot = mat.rotation;            
+            tr.localPosition = pos;
+            tr.localRotation = rot;
+            tr.localScale = scale;
+        }
 
+        void CreateSkinnedMeshRenderer()
+        {
+            Mesh mesh = target as Mesh;
+            if (mesh == null)
+                return;
+            
+            m_BaseGameObjectForSkinnedMeshRenderer = new GameObject {hideFlags = HideFlags.HideAndDontSave};
+            
+            SkinnedMeshRenderer skinnedMeshRenderer = m_BaseGameObjectForSkinnedMeshRenderer.AddComponent<SkinnedMeshRenderer>();
+            skinnedMeshRenderer.hideFlags = HideFlags.HideAndDontSave;
+
+            // setting up a mesh with a proper game object creation chain so it could be destroyed later
```

Comment:
I don't understand what this comment says, or why there need to be two game objects.